lib/identity/client.js

```diff
@@ -886,16 +938,72 @@ class Client extends PartialClient {
       )
     }
     // if the admin sharing group will be empty, revoke the share & delete the group
-    const groupName = `tozny.secret.${this.config.realmName}.${
-      this.storage.config.clientId
-    }.${secret.meta.plain.secretType.toLowerCase()}`
+    const groupName = `tozny.secret.${this.storage.config.clientId}.${this.config.realmName}`
     await this.revokeShareBeforeDeleteSecret(groupName, secret.meta.type)
     // delete the secret
     return await this.storage.deleteRecord(
       secret.meta.recordId,
       secret.meta.version
     )
   }
+  /**
+   * getOrCreateNamespace
+   * @param {Object} namespace
+   * @return {Object} returns the group based on namespace
+   */
+  async getOrCreateNamespace(namespace) {
```

Comment:
This still needs to get refactored out into a private/protected method (removed from living within the client class itself).